packages/table/src/common/internal/selectionUtils.ts

```diff
@@ -0,0 +1,120 @@
+ *      simply expand in the specified direction.
+ *
+ * Other notes:
+ * - A CELLS region can be resized vertically or horizontally.
+ * - A FULL_ROWS region can be resized only vertically.
+ * - A FULL_COLUMNS region can be resized only horizontally.
+ * - A FULL_TABLE region cannot be resized.
+ *
+ * This function does not clamp the indices of the returned region; that is the
+ * responsibility of the caller.
+ */
+export function resizeSelectedRegion(region: IRegion, direction: Direction, focusedCell?: IFocusedCellCoordinates) {
+    if (Regions.getRegionCardinality(region) === RegionCardinality.FULL_TABLE) {
+        // return the same instance to maintain referential integrity and
+        // possibly avoid unnecessary update lifecycles.
+        return region;
+    }
+
+    const nextRegion: IRegion = Regions.copy(region);
+
+    let affectedRowIndex: number = 0;
+    let affectedColumnIndex: number = 0;
+
+    if (focusedCell != null) {
+        const isAtTop = FocusedCellUtils.isFocusedCellAtRegionTop(nextRegion, focusedCell);
+        const isAtBottom = FocusedCellUtils.isFocusedCellAtRegionBottom(nextRegion, focusedCell);
+        const isAtLeft = FocusedCellUtils.isFocusedCellAtRegionLeft(nextRegion, focusedCell);
+        const isAtRight = FocusedCellUtils.isFocusedCellAtRegionRight(nextRegion, focusedCell);
+
+        // the focused cell is found along the top and bottom boundary
+        // simultaneously when the region is 1 row tall. check for this and
+        // similar special cases.
+        if (direction === Direction.UP) {
+            affectedRowIndex = isAtTop && !isAtBottom ? 1 : 0;
+        } else if (direction === Direction.DOWN) {
+            affectedRowIndex = isAtBottom && !isAtTop ? 0 : 1;
+        } else if (direction === Direction.LEFT) {
+            affectedColumnIndex = isAtLeft && !isAtRight ? 1 : 0;
+        } else {
+            // i.e. `Direction.RIGHT:`
+            affectedColumnIndex = isAtRight && !isAtLeft ? 0 : 1;
+        }
+    } else {
+        // when there is no focused cell, expand in the specified direction.
+        affectedRowIndex = direction === Direction.DOWN ? 1 : 0;
+        affectedColumnIndex = direction === Direction.RIGHT ? 1 : 0;
+    }
+
+    const delta = DirectionUtils.directionToDelta(direction);
+
+    if (nextRegion.rows != null) {
+        nextRegion.rows[affectedRowIndex] += delta.rows;
+    }
+    if (nextRegion.cols != null) {
+        nextRegion.cols[affectedColumnIndex] += delta.cols;
+    }
+
+    // the new coordinates might be out of bounds. the caller is responsible for
+    // sanitizing the result.
+    return nextRegion;
+}
+
+/**
+ * Expands the last-selected region to the new region, and replaces the
+ * last-selected region with the expanded region. If a focused cell is provided,
+ * the focused cell will serve as an anchor for the expansion.
+ */
+export function expandLastSelectedRegion(
```

Comment:
In fixing #1706, I'd like to rename this to `expandSelectedRegion` and let the caller determine which region to operate on (based on the `focusedSelectionIndex` etc).

Reply:
Another naming thought for this function: we're effectively creating a minimum bounding box around the focused cell (or if it doesn't exist: the last-selected region) and the provided `newRegion`. Maybe we can name the two functions:

- `getResizedRegion`
- `getBoundingRegion` or `getMinimalBoundingRegion`

?

Reply:
It's also somewhat ambiguous why these functions live in `SelectionUtils` vs. somewhere else. Maybe everything belongs in `Regions.ts`?